conda_build/environ.py

```diff
@@ -47,6 +47,10 @@ def get_git_build_info(src_dir, git_url, expected_rev):
     env = os.environ.copy()
     d = {}
     git_dir = join(src_dir, '.git')
+    if not isinstance(git_dir, str):
```

Comment:
>Could we do something like env['GIT_DIR'] = str(git_dir)

I think that would implicitly use `sys.getdefaultencoding()`, which may or may not be the same thing as `sys.getfilesystemencoding()`.  Since we're dealing with a file path here, which will ultimately be interpreted by the file system, I figured the latter was the correct option.

But at the end of the day, it seems that this is really only a problem on Windows, and who the heck knows if anything makes sense on Windows.